server/router/middleware/session/org.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package session
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+
+	"go.woodpecker-ci.org/woodpecker/v2/server/model"
+	"go.woodpecker-ci.org/woodpecker/v2/server/store"
+	"go.woodpecker-ci.org/woodpecker/v2/server/store/types"
+)
+
+func Org(c *gin.Context) *model.Org {
+	v, ok := c.Get("org")
+	if !ok {
+		return nil
+	}
+	r, ok := v.(*model.Org)
+	if !ok {
+		return nil
+	}
+	return r
+}
+
+func SetOrg() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var (
+			orgID int64
+			err   error
+		)
+
+		orgParam := c.Param("org_id")
+		if orgParam != "" {
+			orgID, err = strconv.ParseInt(orgParam, 10, 64)
+			if err != nil {
+				c.String(http.StatusBadRequest, "Invalid organization ID")
+				c.Abort()
+				return
+			}
+		}
+
+		org, err := store.FromContext(c).OrgGet(orgID)
+		if err != nil {
+			if errors.Is(err, types.RecordNotExist) {
+				c.String(http.StatusNotFound, "Organization not found")
+				c.Abort()
+				return
+			}
+			_ = c.AbortWithError(http.StatusInternalServerError, err)
+			return
+		}
+
+		c.Set("org", org)
+		c.Next()
+	}
+}
+
+func MustOrg() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		SetOrg()(c)
```

Comment:
This seems to be rather implicit. Probably better to explicitly load / set the org in the router as we do it with `SetUser`.

Reply:
?

Must will fail if org could not load and set want fail ... if desired i can remove set for now .... as we have only cades where it must load

Reply:
Works like this as well. I actually wanted to say, that `SetOrg` is kind of a side-effect in a middleware that gets the org and thereby should be explicitly used in the router same as for user. We `SetUser` first to later on be able to get the user through `User()` or `MustUser()`